### ISSUE TITLE
sphinx.writers.html.HTMLTranslator.visit_image(): ResourceWarning

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1375]
`sphinx.writers.html.HTMLTranslator.visit_image()` triggers ResourceWarning when `:scale:` is used.
I attach a patch.

Example (documentation of stevedore-0.14.1):

```
$ PYTHONPATH=".." make html
sphinx-build -b html -d build/doctrees   source build/html
Making output directory...
Running Sphinx v1.2.1
/usr/lib64/python3.3/site-packages/pbr/packaging.py:860: ResourceWarning: unclosed file <_io.TextIOWrapper name='PKG-INFO' mode='r' encoding='UTF-8'>
  version = _get_version_from_pkg_info(package_name)
loading pickled environment... failed: [Errno 2] No such file or directory: '/tmp/stevedore-0.14.1/docs/build/doctrees/environment.pickle'
building [html]: targets for 12 source files that are out of date
updating environment: 12 added, 0 changed, 0 removed
reading sources... [100&#37;] tutorial/testing                                                                                                                   
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
/usr/lib64/python3.3/site-packages/sphinx/writers/html.py:404: ResourceWarning: unclosed file <_io.BufferedReader name='/tmp/stevedore-0.14.1/docs/source/essays/ceilometer-design.jpg'>
  del im
/usr/lib64/python3.3/site-packages/sphinx/writers/html.py:404: ResourceWarning: unclosed file <_io.BufferedReader name='/tmp/stevedore-0.14.1/docs/source/essays/discovery.jpg'>
  del im
/usr/lib64/python3.3/site-packages/sphinx/writers/html.py:404: ResourceWarning: unclosed file <_io.BufferedReader name='/tmp/stevedore-0.14.1/docs/source/essays/enabling.jpg'>
  del im
/usr/lib64/python3.3/site-packages/sphinx/writers/html.py:404: ResourceWarning: unclosed file <_io.BufferedReader name='/tmp/stevedore-0.14.1/docs/source/essays/importing.jpg'>
  del im
/usr/lib64/python3.3/site-packages/sphinx/writers/html.py:404: ResourceWarning: unclosed file <_io.BufferedReader name='/tmp/stevedore-0.14.1/docs/source/essays/integration.jpg'>
  del im
/usr/lib64/python3.3/site-packages/sphinx/writers/html.py:404: ResourceWarning: unclosed file <_io.BufferedReader name='/tmp/stevedore-0.14.1/docs/source/essays/invocation.jpg'>
  del im
/usr/lib64/python3.3/subprocess.py:718: ResourceWarning: unclosed file <_io.BufferedReader name=3>                                                           
  _cleanup()
writing output... [100&#37;] tutorial/testing                                                                                                                    
writing additional files... genindex search
copying images... [100&#37;] essays/invocation.jpg                                                                                                               
copying static files... WARNING: html_static_path entry '/tmp/stevedore-0.14.1/docs/source/_static' does not exist
done
copying extra files... done
dumping search index... done
dumping object inventory... done
build succeeded, 1 warning.

Build finished. The HTML pages are in build/html.
```

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1375
- Originally reported by: Arfrever Frehtes Taifersar Arahesis
- Originally created at: 2014-02-04T04:38:18.343

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2014-03-01 08:08:02+00:00_

Thanks.

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2014-03-01 08:08:04+00:00_

Closes #1375: fix ResourceWarnings with unclosed PIL images.

→ <<cset 048b7d77d3c7192f0ab2768ca536f5d495ebc58b>>
